pkg/api/pluginproxy/pluginproxy.go

```diff
@@ -82,33 +65,9 @@ func NewApiPluginProxy(ctx *models.ReqContext, proxyPath string, route *plugins.
 
 		applyUserHeader(cfg.SendUserHeader, req, ctx.SignedInUser)
 
-		if len(route.Headers) > 0 {
-			headers, err := getHeaders(route, ctx.OrgId, appID)
-			if err != nil {
-				ctx.JsonApiErr(500, "Could not generate plugin route header", err)
-				return
-			}
-
-			for key, value := range headers {
-				log.Tracef("setting key %v value <redacted>", key)
-				req.Header.Set(key, value[0])
-			}
-		}
-
-		if len(route.URL) > 0 {
-			interpolatedURL, err := updateURL(route, ctx.OrgId, appID)
-			if err != nil {
-				ctx.JsonApiErr(500, "Could not interpolate plugin route url", err)
-			}
-			targetURL, err := url.Parse(interpolatedURL)
-			if err != nil {
-				ctx.JsonApiErr(500, "Could not parse custom url: %v", err)
-				return
-			}
-			req.URL.Scheme = targetURL.Scheme
-			req.URL.Host = targetURL.Host
-			req.Host = targetURL.Host
-			req.URL.Path = util.JoinURLFragments(targetURL.Path, proxyPath)
+		if err := AddHeaders(&req.Header, route, data); err != nil {
```

Comment:
Due to other suggestions.

```suggestion
		if err := addHeaders(&req.Header, route, data); err != nil {
```